Scenarios/Contribute.md

```diff
@@ -5,40 +5,36 @@ This section describes how to contribute to AL-Go. How to set up your own enviro
 You can do this in two ways:
 
 - Use a fork of AL-Go for GitHub in your own **personal GitHub account** in development mode
-- Use 3 public repositories in your own **personal GitHub account** (AL-Go-PTE, AL-Go-AppSource and AL-Go-Actions, much like in production)
+- Use 2 public repositories in your own **personal GitHub account** (AL-Go-PTE, AL-Go-AppSource and AL-Go-Actions, much like in production)
 
 ## Use a fork of AL-Go for GitHub in "development mode"
 
 1. Fork the [https://github.com/microsoft/AL-Go](https://github.com/microsoft/AL-Go) repository to your **personal GitHub account**.
 1. You can optionally also create a branch in the AL-Go fork for the feature you are working on.
 
-**https://github.com/<yourGitHubUserName>/AL-Go@<yourBranch>** can now be used as a template in your AL-Go project when running _Update AL-Go System Files_ to use the actions/workflows from this fork.
+**<yourGitHubUserName>/AL-Go@<yourBranch>** can now be used as a template in your AL-Go project when running _Update AL-Go System Files_ to use the actions/workflows from this fork.
```

Comment:
Yes, they can do this - but this requires different skills then setting up some indirect templates.
Plus, it would still only give them a PTE and a AppSource template and it is much harder to maintain a fork of AL-Go than to maintain a set of indirect templates for their various project types.

Reply:
Why is it much harder to maintain a fork of AL-Go than indirect templates? Don't you just need to occasionally run: 
```powershell
git remote add upstream "https://github.com/microsoft/AL-Go"
git pull upstream
```
And then fix the merge conflicts?

Reply:
The requirements are:
1. Be able to create x template repositories for various types of projects. This could be different types of on-premises solutions, PTEs, ISV solutions.
2. Template repositories must be able to be private
3. Template repositories must be able to use AL-Go main, preview or fixed version if necessary.
4.  It should not require deep knowledge of yaml, git, powershell or like to create a new template repository.

There are two ways of using a fork.
You can fork AL-Go or you can fork AL-Go-PTE|AL-Go-AppSource.
If you fork AL-Go-PTE|AL-Go-AppSource, then (1) would have to be in different organizations (you can only have 1 fork in an org). 2 would require a hack and git gymnastics, 3. would be impossible (or at least I don't know how to) and my guess is that you will quickly get into a situation where our changes require you to have more yaml knowledge.

If you fork AL-Go and use direct AL-Go development (using org/AL-Go@branch) then a change to that branch might break your pipelines and even Update AL-Go System Files. 1 would require you to maintain multiple forks. 2 would be hard, 3 would require you to have a preview branch and a main branch with your customizations and then run update in all repositories (potentially 1000s) or do some git gymnastics to update your branch with a different branch.  You would definitely need git skills and yaml skills.

If you fork AL-Go and use Deploy to deploy to your own AL-Go-PTE and your own AL-Go-AppSource (like a complete parallel setup with our AL-Go version), then you are basically developing your own AL-Go version. Having multiple template repositories is not really supported, they cannot be private, it is very hard to maintain a main and a private version and it would definitely require skills that we don't normally require.

Basically this is similar to back in the days, where Navision used source code customizations. Everybody would modify the hell out of the source and would end up not staying up-to-date because it was too hard to follow along.

Today we have an app model with events, which is kind of what this PR implements

You are more than welcome to try and prove me wrong, but by limiting the number of places we support changes, we will keep more people on standard - many things might just be to add some workflows and/or jobs to enhance their experience.

Reply:
> The requirements are:
> 
> 1. Be able to create x template repositories for various types of projects. This could be different types of on-premises solutions, PTEs, ISV solutions.
> 2. Template repositories must be able to be private
> 3. Template repositories must be able to use AL-Go main, preview or fixed version if necessary.
> 4. It should not require deep knowledge of yaml, git, powershell or like to create a new template repository.
> 

@freddydk this is exactly what we would need to make COSMO Alpaca using AL:GO for GitHub
Since we're having a few hundred repos managed by COSMO Alpaca we would also need different Templates. Some of the templates also might be private because they will contain some closed source. 

When managing so many repos, it is absolutely necessary for us to have 100% control over the templates and the version used. With the custom templates we can provide this for our projects.